src/move/rewards/sources/emojicoin_dot_fun_arena.move

```diff
@@ -0,0 +1,614 @@
+        } else {
+            // Move emojicoin 0 balance out of escrow.
+            let emojicoin_0_ref_mut = &mut escrow_ref_mut.emojicoin_0;
+            let input_amount = coin::value(emojicoin_0_ref_mut);
+            aptos_account::deposit_coins(
+                swapper_address, coin::extract_all(emojicoin_0_ref_mut)
+            );
+
+            // Get amount of APT recieved by selling emojicoin 0, then execute swap.
+            let swap_to_apt =
+                emojicoin_dot_fun::simulate_swap<Emojicoin0, EmojicoinLP0>(
+                    swapper_address,
+                    market_address_0,
+                    input_amount,
+                    true,
+                    @integrator,
+                    INTEGRATOR_FEE_RATE_BPS_DUAL_ROUTE
+                );
+            let (_, _, _, _, _, _, _, _, net_proceeds_in_apt, _, _, _, _, _, _, _, _, _) =
+                emojicoin_dot_fun::unpack_swap(swap_to_apt);
+            emojicoin_dot_fun::swap<Emojicoin0, EmojicoinLP0>(
+                swapper,
+                market_address_1,
+                input_amount,
+                true,
+                @integrator,
+                INTEGRATOR_FEE_RATE_BPS_DUAL_ROUTE,
+                1
+            );
+
+            // Get amount of emojicoin 1 recieved by buying it with APT proceeds.
+            let swap_to_emojicoin_1 =
+                emojicoin_dot_fun::simulate_swap<Emojicoin1, EmojicoinLP1>(
+                    swapper_address,
+                    market_address_1,
+                    net_proceeds_in_apt,
+                    false,
+                    @integrator,
+                    INTEGRATOR_FEE_RATE_BPS_DUAL_ROUTE
+                );
+            let (
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                net_proceeds_in_emojicoin_1,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _
+            ) = emojicoin_dot_fun::unpack_swap(swap_to_emojicoin_1);
+            emojicoin_dot_fun::swap<Emojicoin1, EmojicoinLP1>(
+                swapper,
+                market_address_1,
+                net_proceeds_in_apt,
+                false,
+                @integrator,
+                INTEGRATOR_FEE_RATE_BPS_DUAL_ROUTE,
+                1
+            );
+
+            // Move emojicoin 1 balance to escrow.
+            coin::merge(
+                &mut escrow_ref_mut.emojicoin_1,
+                coin::withdraw(swapper, net_proceeds_in_emojicoin_1)
+            );
+        };
+
+        if (exit_once_done)
+            exit_inner<Emojicoin0, EmojicoinLP0, Emojicoin1, EmojicoinLP1>(
+                swapper, melee_id, false
+            );
+    }
+
+    fun init_module(rewards: &signer) {
+        // Get first melee market addresses, without using randomness APIs.
```

Comment:
Is it possible to use randomness APIs during the `init_module` function?

Reply:
Resolving per AUID workaround to https://github.com/aptos-labs/aptos-core/issues/15436